tests/task/test_calcZernikesFitRadius.py

```diff
@@ -0,0 +1,120 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import os
+
+import lsst.utils.tests
+import numpy as np
+from lsst.daf import butler as dafButler
+from lsst.ts.wep.task import (
+    CalcZernikesTask,
+    CalcZernikesTaskConfig,
+)
+from lsst.ts.wep.task.donutStamps import DonutStamps
+from lsst.ts.wep.utils import (
+    getModulePath,
+    runProgram,
+    writeCleanUpRepoCmd,
+    writePipetaskCmd,
+)
+
+
+class TestCalcZernikesFitRadius(lsst.utils.tests.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        """
+        Generate donutCatalog needed for task.
+        """
+        moduleDir = getModulePath()
+        cls.testDataDir = os.path.join(moduleDir, "tests", "testData")
+        testPipelineConfigDir = os.path.join(cls.testDataDir, "pipelineConfigs")
+        cls.repoDir = os.path.join(cls.testDataDir, "gen3TestRepo")
+        cls.runName = "run1"
+
+        # Check that run doesn't already exist due to previous improper cleanup
+        butler = dafButler.Butler(cls.repoDir)
+        registry = butler.registry
+        collectionsList = list(registry.queryCollections())
+        if cls.runName in collectionsList:
+            cleanUpCmd = writeCleanUpRepoCmd(cls.repoDir, cls.runName)
+            runProgram(cleanUpCmd)
+
+        collections = "refcats/gen2,LSSTCam/calib,LSSTCam/raw/all"
+        instrument = "lsst.obs.lsst.LsstCam"
+        cls.cameraName = "LSSTCam"
+        pipelineYaml = os.path.join(
+            testPipelineConfigDir, "testCalcZernikesCwfsSetupPipeline.yaml"
+        )
+
+        pipeCmd = writePipetaskCmd(
+            cls.repoDir, cls.runName, instrument, collections, pipelineYaml=pipelineYaml
+        )
+        pipeCmd += ' -d "detector IN (191, 192)"'
+        runProgram(pipeCmd)
+
+    @classmethod
+    def tearDownClass(cls):
+        cleanUpCmd = writeCleanUpRepoCmd(cls.repoDir, cls.runName)
+        runProgram(cleanUpCmd)
+
+    def setUp(self):
+        self.config = CalcZernikesTaskConfig()
+        self.task = CalcZernikesTask(config=self.config, name="Base Task")
+
+        self.butler = dafButler.Butler(self.repoDir)
+        self.registry = self.butler.registry
+
+        self.dataIdExtra = {
+            "instrument": "LSSTCam",
+            "detector": 191,
+            "exposure": 4021123106000,
+            "visit": 4021123106000,
+        }
+        self.dataIdIntra = {
+            "instrument": "LSSTCam",
+            "detector": 191,
+            "exposure": 4021123106000,
+            "visit": 4021123106000,
+        }
+        self.donutStampsExtra = self.butler.get(
+            "donutStampsExtra", dataId=self.dataIdExtra, collections=[self.runName]
+        )
+        self.donutStampsIntra = self.butler.get(
+            "donutStampsIntra", dataId=self.dataIdExtra, collections=[self.runName]
+        )
+
+    def testEstimateZernikesEmpty(self):
```

Comment:
To test both situations we should add another test where you keep all the donut stamps but set the donut quality table to have all the donuts fail the selection.